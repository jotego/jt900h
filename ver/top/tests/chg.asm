    ; chg num, dst
    main section code
    org 0
    ld a,0xbf    ; common header

    ; chg #4,r
    ld a,0
    chg 0,a
    chg 2,a
    chg 5,a ; a=0xa5
    chg 7,a
    ld  b,a
    ld  c,a
    sub b,0xa5
    jp ne,bad_end
    or ra3,1

    ; chg #3,(mem)
    incf
    ld xix,data
    ld a,(xix)
    chg 1,(xix)
    ld w,(xix)
    decf
    cp (xix),0xfc
    jp ne,bad_end
    or ra3,2


end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end