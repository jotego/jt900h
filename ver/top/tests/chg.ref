000000a5 - 0000fcfe - 00000000 - 00000003
000000a5 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000051 - 00000000 - 00000000 - 00000000
SR = f804
