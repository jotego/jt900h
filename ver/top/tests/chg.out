000000a5 - 00000000 - 00000000 - 00000000
000000a5 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 00000000
SR = f842
